modules/ding_gatewayf/ding_gatewayf.module

```diff
@@ -241,6 +246,70 @@ function ding_gatewayf_callback() {
 }
 
 /**
+ * Do a check on the checksum given from gatewayf.
```

Comment:
But the Ding2 community needs to maintain this going forward so we need to get a shared understanding of what the contract of the implementation is. If there is no documentation to reference then please add a summary of what is going on.

Perhaps something in the lines of:

To protect against malicious attempts to login through gatewayf the response contains a checksum which can be used to verify the data: 

1. Take all recieved values except the checksum
2. Concatenate into a single string
3. Append a shared salt
4. Hash the string using using MD5
5. Compare the hash with the checksum